crates/oxc_semantic/src/builder.rs

```diff
@@ -253,7 +252,9 @@ impl<'a> SemanticBuilder<'a> {
     ) -> SymbolId {
         if let Some(symbol_id) = self.check_redeclaration(scope_id, span, name, excludes, true) {
             self.symbols.union_flag(symbol_id, includes);
-            self.add_redeclared_variables(VariableInfo { name: name.clone(), span, symbol_id });
+            if includes.is_function_scoped_declaration() {
```

Comment:
This rule focuses only on FunctionScopeVariable, e.g., a variable or function parameter defined via `var`.